src/PAModel/SourceTransforms/AppTestTransform.cs

```diff
@@ -43,9 +43,8 @@ public AppTestTransform(ErrorContainer errors, TemplateStore templateStore, Edit
                 _testStepTemplateName = "TestStep" + i;
 
             var idRestorer = new UniqueIdRestorer(entropy);
-            _screenIdToScreenName = stateStore.Contents
-                .Where(state => state.TopParentName == state.Name)
-                .Select(state => new KeyValuePair<string, string>(idRestorer.GetControlId(state.Name).ToString(), state.Name)).ToList();
+            _screenIdToScreenName = app._screens
+                .Select(screen => new KeyValuePair<string, string>(idRestorer.GetControlId(screen.Key).ToString(), screen.Key)).ToList();
```

Comment:
Should the idRestorer be a property on CanvasDocument, if it's tracking global mapping? instead of instantiating it here.

Reply:
We probably can, we use idRestorer at two places currently, one here and the other in IRStateHelper...The IRStateHelper currently doesn't take in CanvasApp object so idRestorer won't be available there, are you suggesting we start passing CanvasApp object to IRStateHelper as well?

Reply:
ignore my comment, I just saw that we are already passing the idRestorer to IRStateHelper, so definitely makes sense for this to be added to CanvasDocument, I ll make the change